ballerina/listener.bal

```diff
@@ -54,7 +54,12 @@ public class Listener {
         if graphiql.enable {
             check validateGraphiqlPath(graphiql.path);
             string gqlServiceBasePath = name is () ? "" : getBasePath(name);
-            HttpService graphiqlService = getGraphiqlService(serviceConfig, gqlServiceBasePath);
+            Engine engine = check new (getSchemaString(serviceConfig), getMaxQueryDepth(serviceConfig));
```

Comment:
Why do we need to create another instance of the `Engine`? This does not seem correct. We have to have only one Engine object.